### PR TITLE
Telescope simulator mountType property

### PR DESCRIPTION
**Pull request by naheedsa**

### CODE REVIEW

File: drivers/telescope/telescope_simulator.cpp [hunk 1]
```diff
@@ -447,13 +446,13 @@ bool ScopeSim::ISNewSwitch(const char *dev, const char *name, ISState *states, c
...
     {
 #ifdef USE_SIM_TAB
-        if (strcmp(name, mountTypeSP.name) == 0)
+        if (strcmp(name, mountTypeSP.getName()) == 0)
         {
-            if (IUUpdateSwitch(&mountTypeSP, states, names, n) < 0)
+            if (IUUpdateSwitch(mountTypeSP, states, names, n) < 0)
                 return false;
 
-            mountTypeSP.s = IPS_OK;
```
Comment (naheedsa) [inline drivers/telescope/telescope_simulator.cpp:455]: there was a confusion of mountTypeSP.s
It should be used as a property?

Comment (knro) [reply drivers/telescope/telescope_simulator.cpp:455]: mountTypeSP.setState(IPS_OK)

File: drivers/telescope/telescope_simulator.cpp [hunk 2]
```diff
@@ -65,11 +65,10 @@ bool ScopeSim::initProperties()
...
     // mount type and alignment properties, these are in the Simulation tab
-    IUFillSwitch(&mountTypeS[Alignment::ALTAZ], "ALTAZ", "AltAz", ISS_OFF);
-    IUFillSwitch(&mountTypeS[Alignment::EQ_FORK], "EQ_FORK", "Fork (Eq)", ISS_OFF);
-    IUFillSwitch(&mountTypeS[Alignment::EQ_GEM], "EQ_GEM", "GEM", ISS_ON);
-    IUFillSwitchVector(&mountTypeSP, mountTypeS, 3, getDeviceName(), "MOUNT_TYPE", "Mount Type",
-                       "Simulation", IP_WO, ISR_1OFMANY, 60, IPS_IDLE );
+    mountTypeSP[0].fill("ALTAZ", "ALTAZ", ISS_OFF);
+    mountTypeSP[1].fill("EQ_FORK", "Fork (Eq)", ISS_OFF);
+    mountTypeSP[2].fill("EQ_GEM", "GEM", ISS_OFF);
+    mountTypeSP.fill(getDeviceName() ,"MOUNT_TYPE", "Mount Type","Simulation", IP_WO, ISR_1OFMANY, 60, IPS_IDLE);
```
Comment (knro) [inline drivers/telescope/telescope_simulator.cpp:71]: Use Enum

File: drivers/telescope/telescope_simulator.cpp [hunk 3]
```diff
@@ -135,8 +134,8 @@ void ScopeSim::ISGetProperties(const char *dev)
     INDI::Telescope::ISGetProperties(dev);
 
 #ifdef USE_SIM_TAB
-    defineProperty(&mountTypeSP);
-    loadConfig(true, mountTypeSP.name);
+    defineProperty(mountTypeSP);
+    loadConfig(true, mountTypeSP->getName());
```
Comment (knro) [inline drivers/telescope/telescope_simulator.cpp:138]: mountTypeSP.load()

File: drivers/telescope/telescope_simulator.cpp [hunk 4]
```diff
@@ -447,13 +446,13 @@ bool ScopeSim::ISNewSwitch(const char *dev, const char *name, ISState *states, c
     if (dev != nullptr && strcmp(dev, getDeviceName()) == 0)
     {
 #ifdef USE_SIM_TAB
-        if (strcmp(name, mountTypeSP.name) == 0)
+        if (strcmp(name, mountTypeSP.getName()) == 0)
```
Comment (knro) [inline drivers/telescope/telescope_simulator.cpp:449]: mountTypeSP.isNameMatch(name))

File: drivers/telescope/telescope_simulator.cpp [hunk 5]
```diff
@@ -447,13 +446,13 @@ bool ScopeSim::ISNewSwitch(const char *dev, const char *name, ISState *states, c
...
         {
-            if (IUUpdateSwitch(&mountTypeSP, states, names, n) < 0)
+            if (IUUpdateSwitch(mountTypeSP, states, names, n) < 0)
```
Comment (knro) [inline drivers/telescope/telescope_simulator.cpp:451]: mountTypeSP.update(states, names, n)

File: drivers/telescope/telescope_simulator.cpp [hunk 6]
```diff
@@ -447,13 +446,13 @@ bool ScopeSim::ISNewSwitch(const char *dev, const char *name, ISState *states, c
...
                 return false;
 
-            mountTypeSP.s = IPS_OK;
-            IDSetSwitch(&mountTypeSP, nullptr);
+            mountTypeSP = IPS_OK;
```
Comment (knro) [inline drivers/telescope/telescope_simulator.cpp]: mountTypeSP.setState(IPS_OK)

File: drivers/telescope/telescope_simulator.cpp [hunk 7]
```diff
@@ -447,13 +446,13 @@ bool ScopeSim::ISNewSwitch(const char *dev, const char *name, ISState *states, c
...
+            IDSetSwitch(mountTypeSP, nullptr);
```
Comment (knro) [inline drivers/telescope/telescope_simulator.cpp:455]: mountTypeSP.apply()

File: drivers/telescope/telescope_simulator.cpp [hunk 8]
```diff
@@ -641,7 +640,7 @@ bool ScopeSim::updateLocation(double latitude, double longitude, double elevatio
 bool ScopeSim::updateMountAndPierSide()
 {
 #ifdef USE_SIM_TAB
-    int mountType = IUFindOnSwitchIndex(&mountTypeSP);
+    int mountType = IUFindOnSwitchIndex(mountTypeSP);
```
Comment (knro) [inline drivers/telescope/telescope_simulator.cpp:643]: int mountType = mountTypeSP.findOnSwitchIndex();

**Pull request closed without merging**